test/libsolidity/util/ContractABIUtils.cpp

```diff
@@ -33,55 +39,256 @@ using namespace dev::solidity::test;
+		_valueTypes.push_back(ABIType{ABIType::Hex});
+	else if (isString(type))
+	{
+		_addressTypes.push_back(ABIType{ABIType::Hex});
 
-	vector<ABIType> abiTypes;
-	if (regex_match(_type, s_boolType))
-		abiTypes.push_back(ABIType{ABIType::Boolean, ABIType::AlignRight, 32});
-	else if (regex_match(_type, s_uintType))
-		abiTypes.push_back(ABIType{ABIType::UnsignedDec, ABIType::AlignRight, 32});
-	else if (regex_match(_type, s_intType))
-		abiTypes.push_back(ABIType{ABIType::SignedDec, ABIType::AlignRight, 32});
-	else if (regex_match(_type, s_bytesType))
-		abiTypes.push_back(ABIType{ABIType::Hex, ABIType::AlignRight, 32});
-	else if (regex_match(_type, s_dynBytesType))
+		if (_isCompoundType)
+			_dynamicTypes.push_back(ABIType{ABIType::Hex});
+
+		_dynamicTypes.push_back(ABIType{ABIType::UnsignedDec});
+		_dynamicTypes.push_back(ABIType{ABIType::String, ABIType::AlignLeft});
+	}
+	else if (isTuple(type))
 	{
-		abiTypes.push_back(ABIType{ABIType::UnsignedDec, ABIType::AlignRight, 32});
-		abiTypes.push_back(ABIType{ABIType::UnsignedDec, ABIType::AlignRight, 32});
-		abiTypes.push_back(ABIType{ABIType::HexString, ABIType::AlignLeft, 32});
+		for (auto const& component: _functionOutput["components"])
+			appendTypesFromName(component, _addressTypes, _valueTypes, _dynamicTypes, true);
 	}
-	else if (regex_match(_type, s_stringType))
+	else if (isBoolArray(type))
+		fill_n(back_inserter(_valueTypes), arraySize(type), ABIType{ABIType::Boolean});
+	else if (isUintArray(type))
+		fill_n(back_inserter(_valueTypes), arraySize(type), ABIType{ABIType::UnsignedDec});
+	else if (isIntArray(type))
+		fill_n(back_inserter(_valueTypes), arraySize(type), ABIType{ABIType::SignedDec});
+	else if (isStringArray(type))
 	{
-		abiTypes.push_back(ABIType{ABIType::UnsignedDec, ABIType::AlignRight, 32});
-		abiTypes.push_back(ABIType{ABIType::UnsignedDec, ABIType::AlignRight, 32});
-		abiTypes.push_back(ABIType{ABIType::String, ABIType::AlignLeft, 32});
+		_addressTypes.push_back(ABIType{ABIType::Hex});
+
+		fill_n(back_inserter(_dynamicTypes), arraySize(type), ABIType{ABIType::Hex});
+
+		for (size_t i = 0; i < arraySize(type); i++)
+		{
+			_dynamicTypes.push_back(ABIType{ABIType::UnsignedDec});
+			_dynamicTypes.push_back(ABIType{ABIType::String, ABIType::AlignLeft});
+		}
+	}
+	else if (isDynBytes(type))
+		return false;
+	else if (isTupleArray(type))
+		return false;
+	else
+		return false;
+
+	return true;
+}
+
+void ContractABIUtils::overwriteParameters(
+	ErrorReporter& _errorReporter,
+	dev::solidity::test::ParameterList& _targetParameters,
+	dev::solidity::test::ParameterList const& _sourceParameters
+)
+{
+	auto overwrite = [&](Parameter _a, Parameter& _b) -> void
+	{
+		if (
+			_a.abiType.size != _b.abiType.size ||
+			_a.abiType.type != _b.abiType.type
+		)
+		{
+			_errorReporter.warning("Type or size of parameter(s) does not match.");
+			_b = _a;
+		}
+	};
+	boost::for_each(_sourceParameters, _targetParameters, boost::bind<void>(overwrite, _1, _2));
```

Comment:
Pull in the lambda function

Reply:
Please check what happens if the two lists do not have the same length.

Reply:
I added an assertion here, because it's not meant to be called with two vectors being unequally sized. In case the number of parameters given in the test does not match the one returned by the ABI, the ones from the ABI are taken and the function has no effect.

But if the expected and the actual ABI parameters have same the same size, but one of the parameters does not match the one defined by the ABI, the function will have an effect and produce that warning.

Reply:
It will loop `min(_source.size(), _target.size())` times.